### PR TITLE
Refactor key machine objects

### PR DESCRIPTION
**Pull request by baude**
In #20538, I was asked to consider refactoring the new OCI pull code from within the generic machine directory.  This is something I had tried when originally coding it but it became apparent that a much larger refactor to prevent circular deps was needed.  Because I did not want to pollute the initial PR with that refactor, I asked for the PR to merge first.  This is the refactor that needed to be done.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/machine/compression/config.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import "strings"
+
+type ImageCompression int64
+
+const (
+	Xz ImageCompression = iota
+	Zip
+	Gz
+	Bz2
+)
```
Comment (TomSweeneyRedHat) [inline pkg/machine/compression/config.go:12]: zstd and/or zstd:chunked?

**Comment by TomSweeneyRedHat** [comment]
Very quick review, LGTM other than one question.  Tests are starting to puke though.

**Comment by jakecorrenti** [comment]
Same question as Tom, but LGTM otherwise

File: pkg/machine/compression/config.go (hunk 1 above)
Comment (baude) [reply pkg/machine/compression/config.go:12]: this is disk compression, so no

**Comment by umohnani8** [comment]
Changes LGTM, needs a rebase though. The sys test failures looks like a flake to me.

**Review by vrothberg — APPROVED** [review]
/lgtm

**Pull request merged**